detekt-rules-exceptions/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ReturnFromFinally.kt

```diff
@@ -45,6 +54,30 @@ class ReturnFromFinally(config: Config = Config.empty) : Rule(config) {
             .forEach { report(CodeSmell(issue, Entity.from(it), issue.description)) }
     }
 
+    override fun visitTryExpression(expression: KtTryExpression) {
+        super.visitTryExpression(expression)
+        if (bindingContext == BindingContext.EMPTY) return
+
+        val finallyBlock = expression.finallyBlock
+
+        if (expression.isUsedAsExpression(bindingContext) && finallyBlock != null) {
```

Comment:
This statement can be switched, since the usage of `bindingContext` in this case could be more expensive.

```suggestion
        if (finallyBlock != null && expression.isUsedAsExpression(bindingContext)) {
```